packages/core/chrome/core-chrome-browser-internal/src/chrome_service.tsx

```diff
@@ -189,6 +190,37 @@ export class ChromeService {
     this.initVisibility(application);
     this.handleEuiFullScreenChanges();
 
+    // Ensure developers are notified if working in a context that lacks the EUI Provider.
+    const isDev = this.params.coreContext.env.mode.name === 'development';
+    if (isDev) {
+      setEuiDevProviderWarning((providerError) => {
+        const errorObject = new Error(providerError.toString());
+        // show a stack trace in the console
+        // eslint-disable-next-line no-console
+        console.error(errorObject);
+
+        notifications.toasts.addDanger({
+          title: '`EuiProvider` is missing',
+          text: mountReactNode(
+            <p data-test-sub="core-chrome-euiDevProviderWarning-toast">
```

Comment:
With this `data-test-subj`, we could catch scenarios when this unwanted toast shows in functional tests, by updating the navigation methods of `CommonPageObject` class: any time there is a navigation in a functional test and this toast message shows, throw a helpful error.

Reply:
Update: https://github.com/elastic/kibana/pull/189018#issuecomment-2274514623